CHANGELOG.md

```diff
@@ -32,7 +40,28 @@ Date: 2023-04-28
 * :recycle: ⚠️ The `master` branch has been renamed to `main`
 * :recycle: The hadron sampler makes now use of the C++17 standard
 
-[Link to diff from previous version](https://github.com/smash-transport/smash/compare/SMASH-hadron-sampler-1.1...SMASH-hadron-sampler-3.0)
+[Link to diff from previous version](https://github.com/smash-transport/smash-hadron-sampler/compare/SMASH-hadron-sampler-1.1...SMASH-hadron-sampler-3.0)
 
 ## SMASH-hadron-sampler-1.1
 Date: 2022-05-17
+
+
+## Release checklist
```

Comment:
I would make the top-level of the list an enumerated list instead of using bullets, `#` instead of `*`. This because steps should be done in order. 😅